alf/algorithms/rl_algorithm.py

```diff
@@ -715,20 +715,35 @@ def train_iter(self):
         else:
             return self._train_iter_off_policy()
 
-    def _train_iter_on_policy(self):
-        """User may override this for their own training procedure."""
-        alf.summary.increment_global_counter()
-
+    @data_distributed_when(lambda algorithm: algorithm.on_policy)
+    def _compute_train_info_and_loss_info_on_policy(self, unroll_length):
         with record_time("time/unroll"):
             with torch.cuda.amp.autocast(self._config.enable_amp):
                 experience = self.unroll(self._config.unroll_length)
             self.summarize_metrics()
 
+        train_info = experience.rollout_info
+        experience = experience._replace(rollout_info=())
+        loss_info = self.calc_loss(train_info)
```

Comment:
From profiling point of view, this used to be under `time/train`. Probably negligible but may deserve letting everyone know about this change.

Reply:
Yes, calc_loss is typically negligible. And we usually don't use on-policy.